### PR TITLE
feat(es/compat): add pure comment for generator and async transform

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc_ecma_transforms_compat/src/es2017/async_to_generator.rs [hunk 1]
```diff
@@ -328,6 +353,9 @@ impl Actual {
 
                 wrapper.function = make_fn_ref(fn_expr);
                 *expr = wrapper.into();
+                if !in_iife && let Some(c) = &mut self.comments {
```
Comment (Austaras) [inline crates/swc_ecma_transforms_compat/src/es2017/async_to_generator.rs:356]: So if let chain arrived, rejoice!

File: crates/swc_ecma_transforms_compat/src/es2015/regenerator/mod.rs [hunk 2]
```diff
@@ -107,10 +113,30 @@ impl Regenerator {
 
         *items = new;
     }
+
+    fn regenerator_mark(&mut self, args: Vec<ExprOrSpread>) -> Expr {
+        let span = Span::dummy_with_cmt();
+
+        if let Some(c) = &mut self.comments {
+            c.add_pure_comment(span.lo)
+        }
```
Comment (magic-akari) [inline crates/swc_ecma_transforms_compat/src/es2015/regenerator/mod.rs:122]:
```suggestion
        let mut span = DUMMY_SP;

        if let Some(c) = &mut self.comments {
            span = Span::dummy_with_cmt();
            c.add_pure_comment(span.lo)
        }
```

**Review by magic-akari — CHANGES_REQUESTED** [review]

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc_ecma_transforms_compat --breaking

**Comment by kdy1** [comment]
CI issues will be fixed by https://github.com/swc-project/swc/pull/5462

**Pull request merged**